clients/go/admin/threelegauth/token_cache_with_keyring.go

```diff
@@ -0,0 +1,61 @@
+package threelegauth
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/flyteorg/flytestdlib/logger"
+
+	"github.com/zalando/go-keyring"
+	"golang.org/x/oauth2"
+)
+
+var defaultCacheProvider = TokenCacheProvider{
```

Comment:
docs for this? Maybe this struct level docs will help to understand that this uses a Keyring based token cache